pkg/generic/thrift/read.go

```diff
@@ -76,8 +77,12 @@ func nextReader(tt descriptor.Type, t *descriptor.TypeDescriptor, opt *readerOpt
 	case descriptor.I64:
 		return readInt64, nil
 	case descriptor.STRING:
-		if t.Name == "binary" && opt.binaryWithBase64 {
-			return readBase64Binary, nil
+		if t.Name == "binary" {
+			if opt.binaryWithByteSlice {
+				return readBinary, nil
```

Comment:
What if both opt.binaryWithByteSlice and opt.binaryWithBase64 are true?

Reply:
@jayantxie I think when both are true, the behavior will be just reading the value and returning []byte, which is the same as when only binaryWithByteSlice is true.

Reply:
I think `opt.binaryWithBase64` is priority to `opt.binaryWithByteSlice` --- no matter the user what string or slice, base64 encoded content should be decoded

Reply:
I misunderstood. Fixed.

Reply:
btw, I found this two option is unlikely to be true at the same time? `opt.binaryWithBase64` for json generic, `opt.binaryWithByteSlice` for map generic

Reply:
@AsterDY binaryWithBase64 is now available for map generic as well